src/dfx/src/commands/deps/init.rs

```diff
@@ -0,0 +1,154 @@
+                            .map_err(|e| anyhow!("Argument is not a valid hex string: {}", e))?;
+                        init_json.set_init_arg(canister_id, None, &bytes);
+                    } else {
+                        let bytes = args_to_bytes(arg_str, &env, &types)?;
+                        init_json.set_init_arg(canister_id, Some(arg_str.to_string()), &bytes);
+                    }
+                }
+                (Some(_), true) => {
+                    bail!("Canister {canister_prompt} takes no init argument. Please rerun without `--argument`");
+                }
+                (None, false) => {
+                    let mut message = format!("Canister {canister_prompt} requires an init argument. The following info might be helpful:");
+                    if let Some(dfx_init) = pulled_json.get_dfx_init(&canister_id)? {
+                        message.push_str(&format!("dfx:init => {dfx_init}"));
+                    }
+                    let candid_args = pulled_json.get_candid_args(&canister_id)?;
+                    message.push_str(&format!("candid:args => {candid_args}"));
+
+                    bail!(message);
+                }
+                (None, true) => {
+                    init_json.set_empty_init(canister_id);
+                }
+            }
+        }
+        None => {
+            let mut canisters_require_init = vec![];
+            for (canister_id, pulled_canister) in &pulled_json.canisters {
+                let canister_prompt = get_canister_prompt(canister_id, pulled_canister);
+                if init_json.contains(canister_id) {
+                    info!(logger, "{canister_prompt} already set init argument.");
+                } else {
+                    let candid_args = pulled_json.get_candid_args(canister_id)?;
+                    let candid_args_idl_types: IDLTypes = candid_args.parse()?;
+                    if candid_args_idl_types.args.is_empty() {
+                        init_json.set_empty_init(*canister_id);
+                    } else {
+                        canisters_require_init.push(canister_prompt);
+                    }
+                }
+            }
+            if !canisters_require_init.is_empty() {
+                let mut message = "The following canister(s) require an init argument. Please run `dfx deps init <NAME/PRINCIPAL>` to set them individually:".to_string();
+                for canister_prompt in canisters_require_init {
+                    message.push_str(&format!("\n{canister_prompt}"));
+                }
+                warn!(logger, "{message}");
+            }
+        }
+    }
+
+    save_init_json(&project_root, &init_json)?;
+    Ok(())
+}
+
+#[context("Failed to validate argument against type defined in candid:args")]
+fn args_to_bytes(
+    arg_str: &str,
+    env: &TypeEnv,
+    types: &[Type], // types has been checked to not be empty
+) -> DfxResult<Vec<u8>> {
+    let first_char = arg_str.chars().next();
```

Comment:
This method looks like it was copy-pasted from [blob_from_arguments](https://github.com/dfinity/sdk/blob/master/src/dfx/src/util/mod.rs#L188).  If so, I'd suggest to extract the method (which may be this exact method) and call it from both places.